packages/cli-repl/src/node-repl-fix-completer-side-effects.ts

```diff
@@ -0,0 +1,73 @@
+import type { CompleterResult } from 'readline';
+import { PassThrough } from 'stream';
+import { promisify } from 'util';
+
+// Detect whether the issue described in https://github.com/nodejs/node/pull/59774
+// occurs in this version of Node.js or not.
+let replHasNodeBug59774: boolean | undefined;
+async function detectReplNodeBug59774(): Promise<boolean> {
+  // 'repl' is not supported in startup snapshots yet.
+  const { start: replStart } =
+    // eslint-disable-next-line @typescript-eslint/consistent-type-imports, @typescript-eslint/no-var-requires
+    require('repl') as typeof import('repl');
+  const repl = replStart({
+    input: new PassThrough(),
+    output: new PassThrough(),
+    terminal: true,
+    useGlobal: false,
+  });
+  let ranFunction = false;
+  repl.context.causeSideEffect = () => (ranFunction = true);
+  try {
+    await promisify(repl.completer)('causeSideEffect().x');
```

Comment:
I'm a bit confused by this, why is it trying to access `.x` in particular? Where is that coming from?